cups/request.c

```diff
@@ -248,6 +248,9 @@ cupsDoIORequest(http_t     *http,	/* I - Connection to server or @code CUPS_HTTP
         (status >= HTTP_STATUS_BAD_REQUEST && status != HTTP_STATUS_UNAUTHORIZED &&
 	 status != HTTP_STATUS_UPGRADE_REQUIRED))
     {
+      if (!errno)
```

Comment:
This would only be for `HTTP_STATUS_ERROR`, so it might be better to update `_cupsSetHTTPError` to accept a `http_t *` argument and look at `http->error` instead of `errno`…

Reply:
Unfortunately I found out we have both use cases in the code - errno is used when we track before we establish connection (in cups/http-addr.c, cups/ipp.c or cups/http.c), so we need to support both :( . I've updated the PR accordingly.

Reply:
Applied.